### PR TITLE
feat(internal): yarn use util/env/docker

### PR DESCRIPTION
**Pull request by rarkins**
Refactors yarn lockfile updating to use our newer util approach.

### CODE REVIEW

File: lib/manager/npm/post-update/yarn.ts [hunk 1]
```diff
@@ -21,103 +20,68 @@ export async function generateLockFile(
...
-      const dedupeCommand =
-        'npx yarn-deduplicate@1.1.1 --strategy fewer && yarn';
-      const dedupeRes = await exec(dedupeCommand, {
-        cwd,
-        env,
-      });
-      // istanbul ignore next
-      stdout += dedupeRes.stdout || '';
-      stderr += dedupeRes.stderr || '';
+      commands.push('npx yarn-deduplicate@2.0.0 --strategy fewer');
```
Comment (viceice) [inline lib/manager/npm/post-update/yarn.ts:65]: I thinnk we should make the version configurable, or at least as const in module scope

File: lib/workers/branch/lock-files/yarn.spec.ts [hunk 2]
```diff
@@ -27,7 +28,7 @@ describe(getName(__filename), () => {
     jest.resetAllMocks();
     env.getChildProcessEnv.mockReturnValue(envMock.basic);
   });
-  it.each([['1.0.0']])(
+  it.each([['1.22.0']])(
```
Comment (viceice) [inline lib/workers/branch/lock-files/yarn.spec.ts:31]: needs some cleanup later here

File: lib/manager/npm/post-update/yarn.ts (hunk 1 above)
Comment (rarkins) [reply lib/manager/npm/post-update/yarn.ts:65]: What's better for now - a fixed version or no version at all?

Comment (viceice) [reply lib/manager/npm/post-update/yarn.ts:65]: I think better no version at all and make it configurable later

Comment (viceice) [inline lib/manager/npm/post-update/yarn.ts:65]:
```suggestion
      commands.push('npx yarn-deduplicate --strategy fewer');
```

**Review by viceice — CHANGES_REQUESTED** [review]

**Review by viceice — DISMISSED** [review]

File: lib/manager/npm/post-update/yarn.ts [hunk 3]
```diff
@@ -21,101 +20,68 @@ export async function generateLockFile(
...
-    const installCmd = cmd + ' install' + cmdExtras;
-    // TODO: Switch to native util.promisify once using only node 8
-    await exec(installCmd, {
+    const execOptions: ExecOptions = {
       cwd,
-      env: cmdEnv,
-    });
+      extraEnv: {
+        NPM_CONFIG_CACHE: process.env.NPM_CONFIG_CACHE,
+        npm_config_store: process.env.npm_config_store,
```
Comment (viceice) [inline lib/manager/npm/post-update/yarn.ts:34]: Why we don't use the passed env?

Comment (rarkins) [reply lib/manager/npm/post-update/yarn.ts:34]: How?

Comment (viceice) [reply lib/manager/npm/post-update/yarn.ts:34]: See passed `env` function argument.

**Review by viceice — APPROVED** [review]

**Pull request merged**